framework/src/userobject/CoupledVarThresholdElementSubdomainModifier.C

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledVarThresholdElementSubdomainModifier.h"
+
+#include "libmesh/quadrature.h"
+
+registerMooseObject("MooseApp", CoupledVarThresholdElementSubdomainModifier);
+
+InputParameters
+CoupledVarThresholdElementSubdomainModifier::validParams()
+{
+  InputParameters params = ThresholdElementSubdomainModifier::validParams();
+  params.addRequiredCoupledVar("coupled_var",
+                               "Coupled variable whose value is used in the criterion");
+  return params;
+}
+
+CoupledVarThresholdElementSubdomainModifier::CoupledVarThresholdElementSubdomainModifier(
+    const InputParameters & parameters)
+  : ThresholdElementSubdomainModifier(parameters), _v(coupledValue("coupled_var"))
+{
+}
+
+Real
+CoupledVarThresholdElementSubdomainModifier::computeValue()
+{
+  Real avg_val = 0;
+
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+    avg_val += _v[qp];
```

Comment:
should this be multiplied by `_JxW[_qp] * _coord[_qp]`?

Reply:
Good catch! I fixed that. @dewenyushu this will probably improve some of your AM results.

Reply:
It is indeed a good catch!!

Reply:
Oops I guess so! I had trouble with element activation using coupled variable. Maybe this is one of the reasons! Nice catch!